app/views/search/_view_all_link.html.erb

```diff
@@ -3,7 +3,7 @@
   <% if @results['total'] > 0 %>
     <% if Flipflop.enabled?(:local_browse) && @results['local_view_more'] %>
       <% view_more_url = @results['local_view_more'] %>
-    <% elsif Flipflop.enabled?(:primo_search) %>
+    <% elsif Flipflop.enabled?(:primo_search) && (params[:target] == ENV.fetch('PRIMO_BOOK_SCOPE') || params[:target] == ENV.fetch('PRIMO_ARTICLE_SCOPE')) %>
       <% view_more_url = @results['primo_ui_view_more'] %>
     <% else %>
       <% view_more_url = @results['eds_ui_view_more'] %>
```

Comment:
I... may have some words to say one day about our variable naming on the final `else` condition lol.

Reply:
@JPrevost Agreed. It felt out of scope here since shipping this is fairly time-sensitive, but we can and will fix that when we rip out EDS.